crates/katana-core/src/starknet/mod.rs

```diff
@@ -240,10 +256,18 @@ impl StarknetWrapper {
         .map_err(|e| e.into())
     }
 
-    // Returns the StarknetState of the underlying Starknet instance.
-    #[allow(unused)]
-    fn state(&self) -> &DictStateReader {
-        unimplemented!("StarknetWrapper::state")
+    pub fn get_state(&self, block_number: BlockNumber) -> Option<DictStateReader> {
```

Comment:
```suggestion
    pub fn state(&self, block_number: BlockNumber) -> Option<DictStateReader> {
```

it's a nit, but rust convention is to not prefix getters with `get_`